R/create_burden_templates.R

```diff
@@ -0,0 +1,537 @@
+##'
+##' @param files filenames, it has to contain the following three files
+##' "Country_Disease_Table.csv", "model_meta.csv" and "model_outcomes.csv"
+##'
+##' @param files_path the directory that contains those files above
+##'
+##' @param assert_files needed meta files
+##' "Country_Disease_Table.csv", "model_meta.csv" and "model_outcomes.csv"
+##'
+##' @param central_templates_only TRUE if you only need central burden templates, usually for open call. But it depend on Tini's input.
+##'  Set to FALSE as default, which outputs both central and stochastic templates
+##'
+##' @export
+create_burden_template <- function(files, files_path, assert_files = c("Country_Disease_Table.csv", "model_meta.csv", "model_outcomes.csv"), central_templates_only = FALSE) {
+  assert_has_files(files, assert_files)
+  files <- paste0(files_path, files)
+
+  country_disease_table <- read_csv(files[1])
+  model_meta_data <- read_csv(files[2])
+  model_outcomes <- read_csv(files[3])
+
+  names(country_disease_table)[names(country_disease_table) == "Country"] <-
+    "country_name"
+  names(country_disease_table)[names(country_disease_table) == "ISO3.code"] <-
+    "country"
+
+  ## making sure that the final year makes sense:
+  year_end <- 2030
+  model_meta_data$year_max <- apply(cbind(2100, year_end + model_meta_data$burden_max_age), 1, min)
+  ## write templates
+  output_files <- sapply(seq_along(model_meta_data[, 1]),
+                         function(i) write_template_files(model_meta_data[i, ],
+                                                          year_end,
+                                                          country_disease_table,
+                                                          model_outcomes,
+                                                          central_templates_only))
+}
+
+### From Nick's insert_function.R
+# function adds new rows to dataframe with contents of value_insert
+# to the column column_name
+
+insert_dataframe <- function(data,column_name,value_insert) {
+
+  length_values <- length(value_insert)
+
+  data$column_name <- value_insert[1]
+
+  cf <- "column_name"
+  names(data)[names(data)%in%cf] <- c(column_name)
+
+  data1 <- data
+
+  for(i in 2:length_values) {
+    data1[, column_name] <- value_insert[i]
+    data <- rbind(data,data1)
+  }
+
+  return(data)
+
+}
+
+# inserts country name into every line of data frame for each country iso code
+
+country_function <- function(k, country_n, country_ISOcode, country_data) {
+
+  country_name_rep <- rep(country_n[k],length(country_data[country_data$country == country_ISOcode[k], ]$country))
+  return(country_name_rep)
+
+}
+
+# outer wrapper function that produces the csv files
+# that selects the models associated with each disease
+
+model_select_function <- function(j) {
+
+  model_name <- unique(model_meta_data[model_meta_data$disease == disease_name[j], ]$modelling_group)
+
+  sapply(1:length(model_name), generate_csv, model_name, j)
```

Comment:
`generate_csv` returns nothing so don't use csv, use a for loop.  Avoid `1:length(foo)` in favour of `seq_along(foo)`